lms/services/exceptions.py

```diff
@@ -113,7 +113,10 @@ def raise_from(cause):
         exception_class = CanvasAPIServerError
 
         if status_code == 401:
-            exception_class = CanvasAPIAccessTokenError
+            if "status" in response.json():
```

Comment:
Can you clarify how the presence of a `status` field in the response allows us to distinguish between the two categories of error? I can make an educated guess, but I'd like to be sure. Are there any scenarios you can think of where this might produce an incorrect result?

Reply:
Just based on testing the Canvas API by calling it various ways the absence of a `"status"` field seems to be the only thing we can use to distinguish access token errors from other 401 errors. See this Slack thread for notes: https://hypothes-is.slack.com/archives/C4K6M7P5E/p1604492209058900

I'm not aware of any non-access token 401 responses from Canvas that lack a `status` field, but there's no way to know for sure.

There's also a risk that Canvas could add a `status` to access token errors in the future.

But I don't think we have any choice -- this is the only that I could fine to way to distinguish them

Reply:
OK, I think it would be good to note this uncertainty in a comment. Let's roll with it for now.